### PR TITLE
Add configurable code to flush the network socket upon error.

### PR DESCRIPTION
**Pull request by sarenameas**
Flush the network socket upon error when configurable macro HTTP_FLUSH_NETWORK_SOCKET_ON_ERROR is set to 1U.
Use the user HTTPResponse_t.pBuffer to flush the network socket.

If the there are too many headers in the response, the response cannot fit in the configured pResponse->pBuffer, or an error occurred during parsing, then there could be data left on the socket. If the application desires continuing to use same connection after an error is returned, the socket needs to be flushed of older data.

### CODE REVIEW

File: source/core_http_client.c [hunk 1]
```diff
@@ -2019,6 +2020,25 @@ static HTTPStatus_t receiveAndParseHttpResponse( const TransportInterface_t * pT
                                                pResponse->bufferLen );
     }
 
+    /* If there was an error, then flush any possible data on the network
+     * socket. If there was an error the response buffer is invalid anyways, so
+     * the response buffer is used to flush the socket. */
+    #if ( HTTP_FLUSH_NETWORK_SOCKET_ON_ERROR == 1U )
```
Comment (aggarw13) [inline source/core_http_client.c:2026]: A unit test for this logic?

File: source/include/core_http_client.h [hunk 2]
```diff
@@ -700,6 +700,8 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * body in @p pRequestBodyBuf over the transport. The response is received in
  * #HTTPResponse_t.pBuffer.
  *
+ * If this function returns an error any data in @p pResponse is invalid.
```
Comment (aggarw13) [inline source/include/core_http_client.h:703]:
```suggestion
 * If this function returns an error, any data in @p pResponse is invalid.
```

File: source/include/core_http_client.h [hunk 3]
```diff
@@ -708,6 +710,12 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * #HTTP_MAX_CONTENT_LENGTH_HEADER_LENGTH for the maximum Content-Length header
  * field and value that could be written to the buffer.
  *
+ * By default HTTP_FLUSH_NETWORK_SOCKET_ON_ERROR is set to 1. Therefore, upon
+ * any error in parsing the response, the network socket is flushed using the
```
Comment (aggarw13) [inline source/include/core_http_client.h:714]:
```suggestion
 * any error in parsing the response, the network socket is flushed of any remaining data using the
```

File: source/core_http_client.c (hunk 1 above)
Comment (sarenameas) [reply source/core_http_client.c:2026]: Still have 100% coverage

File: source/include/core_http_client.h (hunk 3 above)
Comment (sarenameas) [reply source/include/core_http_client.h:714]: Thanks for your review. I think I may be closing this PR and adding the logic with the send and recv timeout.

**Pull request closed without merging**